src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -1071,6 +1083,17 @@ bool ReplicatedMergeTreeQueue::shouldExecuteLogEntry(
             return false;
         }
 
+        auto replica_to_execute_merge = merge_strategy_picker.pickReplicaToExecuteMerge(entry);
+
+        if (replica_to_execute_merge && !merge_strategy_picker.isMergeFinishedByReplica(replica_to_execute_merge.value(), entry))
+        {
+            String reason = "Not executing merge for the part " + entry.new_part_name
+                +  ", waiting for " + replica_to_execute_merge.value() + " to execute merge.";
+            LOG_DEBUG(log, reason);
```

Comment:
Actually that can create quite a lot of logging (it check shouldExecuteLogEntry really often).